src/pages/iou/MoneyRequestCreatedPage.js

```diff
@@ -0,0 +1,128 @@
+        comment: '',
+        created: '',
+        participants: [],
+    },
+};
+
+class MoneyRequestCreatedPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.updateDate = this.updateDate.bind(this);
+        this.navigateBack = this.navigateBack.bind(this);
+        this.iouType = lodashGet(props.route, 'params.iouType', '');
+        this.reportID = lodashGet(props.route, 'params.reportID', '');
+    }
+
+    componentDidMount() {
+        const moneyRequestId = `${this.iouType}${this.reportID}`;
+        const shouldReset = this.props.iou.id !== moneyRequestId;
+        if (shouldReset) {
+            IOU.resetMoneyRequestInfo(moneyRequestId);
+        }
+
+        if (_.isEmpty(this.props.iou.participants) || (this.props.iou.amount === 0 && !this.props.iou.receiptPath) || shouldReset) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+        }
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    componentDidUpdate(prevProps) {
+        // ID in Onyx could change by initiating a new request in a separate browser tab or completing a request
+        if (_.isEmpty(this.props.iou.participants) || (this.props.iou.amount === 0 && !this.props.iou.receiptPath) || prevProps.iou.id !== this.props.iou.id) {
+            // The ID is cleared on completing a request. In that case, we will do nothing.
+            if (this.props.iou.id) {
+                Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+            }
+        }
+    }
+
+    navigateBack() {
+        Navigation.goBack(ROUTES.getMoneyRequestConfirmationRoute(this.iouType, this.reportID));
+    }
+
+    /**
+     * Sets the money request comment by saving it to Onyx.
+     *
+     * @param {Object} value
+     * @param {String} value.moneyRequestCreated
+     */
+    updateDate(value) {
+        IOU.setMoneyRequestCreated(value.moneyRequestCreated);
+        this.navigateBack();
+    }
+
+    render() {
+        return (
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                shouldEnableMaxHeight
+            >
+                <HeaderWithBackButton
+                    title={this.props.translate('common.date')}
+                    onBackButtonPress={this.navigateBack}
+                />
+                <Form
+                    style={[styles.flexGrow1, styles.ph5]}
+                    formID={ONYXKEYS.FORMS.MONEY_REQUEST_CREATED_FORM}
+                    onSubmit={this.updateDate}
+                    submitButtonText={this.props.translate('common.save')}
+                    enabledWhenOffline
+                >
+                    <NewDatePicker
+                        inputID="moneyRequestCreated"
+                        label={this.props.translate('common.date')}
+                        defaultValue={this.props.iou.created}
+                        maxDate={new Date()}
```

Comment:
1. Should we save drafts?
2. What happens when we there is a server error? Are we following Red Brick Road pattern here?

Reply:
No when you close the modal, it should reset that is intentional. RBR is handled separately